src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -2066,55 +2051,48 @@ private void ValidateOrganizationUpgradeParameters(Models.StaticStore.Plan plan,
             }
         }
 
-        private async Task ValidateOrganizationUserUpdatePermissionsAsync(Guid loggedInUserId, Guid organizationId,
-            OrganizationUserType newType, OrganizationUserType? oldType)
+        private void ValidateOrganizationUserUpdatePermissions(Guid organizationId, OrganizationUserType newType,
+            OrganizationUserType? oldType)
         {
-            var loggedInUserOrgs = await _organizationUserRepository.GetManyByUserAsync(loggedInUserId);
-            var loggedInAsOrgOwner = loggedInUserOrgs
-                .Any(u => u.OrganizationId == organizationId && u.Type == OrganizationUserType.Owner);
-            if (loggedInAsOrgOwner)
+            if (_currentContext.OrganizationOwner(organizationId))
             {
                 return;
             }
 
             var isOwner = oldType == OrganizationUserType.Owner;
             var nowOwner = newType == OrganizationUserType.Owner;
-            var ownerUserConfigurationAttempt = (isOwner && nowOwner) || !(isOwner.Equals(nowOwner));
+            var ownerUserConfigurationAttempt = (isOwner && nowOwner) || !isOwner.Equals(nowOwner);
             if (ownerUserConfigurationAttempt)
             {
                 throw new BadRequestException("Only an Owner can configure another Owner's account.");
             }
 
-            var loggedInAsOrgAdmin = loggedInUserOrgs.Any(u => u.OrganizationId == organizationId && u.Type == OrganizationUserType.Admin);
-            if (loggedInAsOrgAdmin)
+            if (_currentContext.OrganizationAdmin(organizationId))
             {
                 return;
             }
 
             var isCustom = oldType == OrganizationUserType.Custom;
             var nowCustom = newType == OrganizationUserType.Custom;
-            var customUserConfigurationAttempt = (isCustom && nowCustom) || !(isCustom.Equals(nowCustom));
+            var customUserConfigurationAttempt = (isCustom && nowCustom) || !isCustom.Equals(nowCustom);
             if (customUserConfigurationAttempt)
             {
                 throw new BadRequestException("Only Owners and Admins can configure Custom accounts.");
             }
 
-            var loggedInAsOrgCustom = loggedInUserOrgs.Any(u => u.OrganizationId == organizationId && u.Type == OrganizationUserType.Custom);
-            if (!loggedInAsOrgCustom)
+            if (!_currentContext.OrganizationCustom(organizationId))
             {
                 return;
```

Comment:
This feels a bit off, feels like we should throw an error here. The user is not owner, nor admin, nor custom. Hence they should not be able to modify users?

Reply:
I'm not sure this check is necessary at all... we further down check for manage users permissions so why it would need to check custom here seems moot, but otherwise, yes, you're correct, throwing an exception seems more appropriate, although redundant with the check below it.

Reply:
I removed the check, as you say it shouldn't be needed. And seems better to error out than somehow accepting it due to missing checks prior.

Reply:
On a second note, Seems we use the function in the import API. Will investigate it a bit more tomorrow to get a better understanding if we can handle permissions differently for those requests.